### DISCUSSION TITLE
How to return a String created in the code?

### DISCUSSION THREAD

**Discussion by szabgab** [rwf2/Rocket#2693] [Questions] [answered]
While experimenting with Rocket I tried to send back a generated string.

```rust
#[macro_use]
extern crate rocket;

#[get("/")]
fn index() -> &'static str {
    let answer = 42;
    format!("The answer is {}", answer)
    //format!("The answer is {}", answer).as_str()
}

#[launch]
fn rocket() -> _ {
    rocket::build().mount("/", routes![index])
}
```

I got an error:

```rust
expected `&str`, found `String`
```

I tried to call `as_str()` and then I got

```
returns a value referencing data owned by the current function
```

I could use a Template, but I wonder if there is a way to send back a string.

I could also define the return to be `String` and then this would work, but the in examples I saw on the [Rocket](https://rocket.rs/) site the examples return `&str`. Does returning `String` have issues?

```rust
#[macro_use]
extern crate rocket;

#[get("/")]
fn index() -> String {
    let count = 42;
    format!("The answer is {}", count)
}

#[launch]
fn rocket() -> _ {
    rocket::build().mount("/", routes![index])
}
```

**Comment by SergioBenitez** [comment, marked as answer]
No, that's perfectly correct. You cannot return a reference to something that heap allocated on the outgoing frame; that would be a dangling pointer.
